modules/tinymce/src/core/main/ts/content/GetContentImpl.ts

```diff
@@ -22,9 +22,14 @@ const getContentFromBody = (editor: Editor, args: GetContentArgs, body: HTMLElem
   if (args.format === 'raw') {
     content = Tools.trim(TrimHtml.trimExternal(editor.serializer, body.innerHTML));
   } else if (args.format === 'text') {
-    // return empty string for text format when editor is empty to avoid bogus elements being returned in content
-    content = Zwsp.trim(body.innerText);
-    content = content === '\n' ? '' : content;
+    content = body.innerHTML.replace(/<br data-mce-bogus="1">/g, '\uFEFF');
+    const parseBody = document.createElement('body');
```

Comment:
We should not be creating elements from, nor appending to the root document. We should be using the editor document instead so please swap the uses here to use `editor.getDoc()` instead (ideally that should be stored as a variable). This is also needed to ensure any content css/styling is applied to the elements when we're getting the text content.